packages/komencikit/src/kit.ts

```diff
@@ -0,0 +1,378 @@
+    if (!signatureResp.ok) {
+      return signatureResp
+    }
+
+    const payload: StartSessionPayload = {
+      externalAccount: this.externalAccount,
+      captchaResponseToken: captchaToken,
+      signature: signatureResp.result,
+    }
+
+    const resp = await this.client.exec(startSession(payload))
+
+    if (resp.ok) {
+      this.client.setToken(resp.result.token)
+      return Ok(resp.result.token)
+    } else if (resp.error.errorType === FetchErrorTypes.Unauthorised) {
+      return Err(new AuthenticationFailed())
+    }
+
+    return resp
+  }
+
+  /**
+   * getDistributedBlindedPepper: uses the /v1/distributedBlindedPepper endpoint to
+   * request the identifier and pepper associated with a phone number in a fee-less scenario
+   *
+   * @param e164Number - phone number
+   * @param clientVersion
+   * @returns the identifier and the pepper
+   */
+  getDistributedBlindedPepper = async (
+    e164Number: string,
+    clientVersion: string
+  ): Promise<Result<GetDistributedBlindedPepperResp, FetchError>> => {
+    return this.client.exec(getDistributedBlindedPepper({ e164Number, clientVersion }))
+  }
+
+  /**
+   * deployWallet: uses the /v1/deployWallet endpoint to deploy a MetaTransactionWallet Proxy
+   * pointing it to the implementation passed as an argument
+   * The function takes care of waiting for retrying, waiting for receipt and log parsing
+   *
+   * @param implementationAddress the implementation address Valora requires
+   * @returns the wallet address
+   */
+  @retry({
+    tries: 3,
+    bailOnErrorTypes: [
+      FetchErrorTypes.Unauthorised,
+      FetchErrorTypes.ServiceUnavailable,
+      TxErrorTypes.Revert,
+    ],
+    onRetry: (_args, error, attempt) => {
+      console.debug(`${TAG}/deployWallet attempt#${attempt} error: `, error)
+    },
+  })
+  public async deployWallet(
+    implementationAddress: string
+  ): Promise<Result<string, FetchError | TxError>> {
+    const resp = await this.client.exec(deployWallet({ implementationAddress }))
+    if (resp.ok) {
+      if (resp.result.status === 'deployed') {
+        return Ok(resp.result.walletAddress)
+      } else {
+        const txHash = resp.result.txHash
+        const receiptResult = await this.waitForReceipt(txHash)
+        if (receiptResult.ok) {
+          const receipt = receiptResult.result
+          if (!receipt.status) {
+            // TODO: Possible to extract reason?
+            return Err(new TxRevertError(txHash, ''))
+          }
+
+          const deployer = await this.contractKit.contracts.getMetaTransactionWalletDeployer(
+            resp.result.deployerAddress
+          )
+          const events = await deployer.getPastEvents('WalletDeployed', {
+            fromBlock: receipt.blockNumber,
+            toBlock: receipt.blockNumber,
+          })
+
+          const deployWalletLog = events.find(
+            (event) => normalizeAddressWith0x(event.returnValues.owner) === this.externalAccount
```

Comment:
We should also verify that the implementationAddress in the event matches the one sent to the API. This prevents a compromised Komenci node from deploying a malicious MTWProxy instance and Valora using it as its own.